spec/features/user_deletes_registered_application_spec.rb

```diff
@@ -13,7 +13,26 @@
 
     click_link "Delete Registered Application"
 
+    page.driver.browser.switch_to.alert.accept
+
     expect(page).to have_content("You have not registered any applications yet.")
     expect(page).to have_css('.flash', text: 'Application deleted successfully.')
   end
+
+  scenario "User Cancels" do
+    user = create(:user)
+    application = create(:application, user: user)
+
+    login_user(user)
+
+    click_link application.name
+
+    expect(page).to have_title(application.name)
+
+    click_link "Delete Registered Application"
+
+    page.driver.browser.switch_to.alert.dismiss
```

Comment:
Good job on getting alert dismissing working.

```ruby
page.driver.browser.switch_to.alert.dismiss
```

The line works (which is the main thing!), however it has a couple (minor) issues:

- it reveals unnecessary/distracting details about the driver
- it is tightly coupled to the browser driver implementation in use
- it may not work if a different browser driver were used

Consider trying the `accept_alert` which is a more concise and driver-independent method provided by Capybara:

```ruby
accept_alert('text that appears in the alert') do
  click_link "Delete Registered Application"
end
```
API docs: http://www.rubydoc.info/gems/capybara/Capybara%2FSession%3Aaccept_alert
README docs on modals: https://github.com/jnicklas/capybara#modals